src/app/actions/actions.component.ts

```diff
@@ -19,20 +19,24 @@ export class ActionsComponent implements OnInit {
 
   ngOnInit() {
     this.userActions = this.userActionsService.getUserActions();
-    this.userActionsService.addDefaultAction(this.selectedAction, this.matchTypes[0]);
-    this.userActionsService.addDefaultCondition(this.matchTypes[0]);
+    this.userActionsService.addDefaultAction(this.selectedAction, this.actionMatchTypes[0]);
+    this.userActionsService.addDefaultCondition(this.conditionMatchTypes[0]);
   }
 
-  get matchTypes(): string[] {
-    return this.appGlobalsService.matchTypes;
+  get actionMatchTypes(): string[] {
+    return this.appGlobalsService.actionMatchTypes;
+  }
+
+  get conditionMatchTypes(): string[] {
+    return this.appGlobalsService.conditionMatchTypes;
   }
 
   addAction() {
-    this.userActionsService.addDefaultAction(this.selectedAction, this.matchTypes[0]);
+    this.userActionsService.addDefaultAction(this.selectedAction, this.actionMatchTypes[0]);
```

Comment:
why you are passing a selected action and default match type to a method which is called `addDefaultAction` this picking the default should be inside of this function.